yt/frontends/gadget/data_structures.py

```diff
@@ -189,6 +216,39 @@ def _calculate_offsets(self, field_list, pcounts):
             self._file_size,
         )
 
+    def _field_count(self, ptype, field):
+        # get the expected array size of the current particle type and field
+        count = self.total_particles[ptype]
+        if field in self.io._vector_fields:
```

Comment:
I think in this particular case, we should pass these attributes down from the `io` object at initialization.

Reply:
agreed! should I add to this PR or hold off for more general refactor? There are other similar cases, so could address them all in one PR.

Reply:
Wait and do it later.